File: programmers/01.hash/02.폰켓몬/python/01.py
def solution(nums):
    set_list = set(nums)
    if len(set_list) > len(nums)/2:
        return int(len(nums)/2)
    else :
        return len(set_list)


print(solution([3,1,2,3]))
print(solution([3,3,3,2,2,4]))
print(solution([3,3,3,2,2,2]))

File: programmers/01.알고리즘/01.hash/02.폰켓몬/js/01.js
function solution(nums) {
    const setArr = new Set(nums);
    return setArr.size > nums.length / 2 ? nums.length / 2 : setArr.size;
}

console.log(solution([3, 1, 2, 3]));
console.log(solution([3, 3, 3, 2, 2, 4]));
console.log(solution([3, 3, 3, 2, 2, 2]));
